docs/faq/add_sparse_op_in_backend.md

```diff
@@ -0,0 +1,439 @@
+.set_attr<nnvm::FInferType>("FInferType", QuadraticOpType)
+.set_attr<FInferStorageType>("FInferStorageType", QuadraticOpStorageType)
+.set_attr<FCompute>("FCompute<cpu>", QuadraticOpForward<cpu>)
+.set_attr<FComputeEx>("FComputeEx<cpu>", QuadraticOpForwardEx<cpu>)
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseIn{"_backward_quadratic"})
+.set_attr<nnvm::FInplaceOption>("FInplaceOption",
+  [](const NodeAttrs& attrs) {
+    return std::vector<std::pair<int, int> >{{0, 0}};
+  })
+.add_argument("data", "NDArray-or-Symbol", "Input ndarray")
+.add_arguments(QuadraticParam::__FIELDS__());
+```
+
+If you compare it with the original registration code,
+only three lines of code are added to the above code block:
+- MXNET_ADD_SPARSE_OP_ALIAS(quadratic)
+This line adds an alias for the quadratic function in
+python frontend so that `quadratic` is accessible from both `mx.symbol.sparse`
+and `mx.ndarray.sparse`.
+- .set_attr<FInferStorageType>("FInferStorageType", QuadraticOpStorageType)
+This line registers the storage type inference attribute of the operator.
+- .set_attr<FComputeEx>("FComputeEx<cpu>", QuadraticOpForwardEx<cpu>)
+This line registers the `FComputeEx` attribute of the operator.
+
+To register this sparse operator on GPU, `quadratic_op.cu` is extended
+as below:
+```cpp
+NNVM_REGISTER_OP(quadratic)
+.set_attr<FCompute>("FCompute<gpu>", QuadraticOpForward<gpu>)
+.set_attr<FComputeEx>("FComputeEx<gpu>", QuadraticOpForwardEx<gpu>);
+```
+
+### Unit Test
+To unit test the sparse operator in frontend, we need to add the following code
+to the python file `test_sparse_ndarray.py`. 
+```python
+@with_seed()
+def test_sparse_quadratic_function():
+    def f(x, a, b, c):
+        return a * x**2 + b * x + c
+
+    def check_sparse_quadratic_function(c):
+      # check forward and compare the result with dense op
+      ndim = 2
+      shape = rand_shape_nd(ndim, 5)
+      data = rand_ndarray(shape=shape, stype='csr')
+      data_np = data.asnumpy()
+      expected = f(data_np, a, b, c)
+      output = mx.nd.sparse.quadratic(data, a=a, b=b, c=c)
+      assert(output.stype == expected_stype)
+      assert_almost_equal(output.asnumpy(), expected)
+
+    a = np.random.random_sample()
+    b = np.random.random_sample()
+    check_sparse_quadratic_function(0.0, 'csr')
+    check_sparse_quadratic_function(1.0, 'default')
+
+```
+
+In this test, we are testing the result of the `sparse.quadratic` operator
+on two cases:
+- CSRNDArray input with c = 0.0, which outputs a CSRNDArray
+- CSRNDArray input with c = 1.0, which outputs a NDArray
+
+## Backward Function
+So far, only the forward operator supports sparse inputs. To add sparse support to the
+backward operator, you also need to register these two attributes to `_backward_quadratic`:
+- `FComputeEx` for sparse operator implementation
+- `FInferStorage` for storage type inference in the backward computation.
+Due to length constraint, this is left as an exercise for readers.
```

Comment:
Maybe write another tutorial on that? Having more ground truth implementations may increase number of contributions.